src/transform.js

```diff
@@ -0,0 +1,41 @@
+import { properties as childProperties } from "./children.js";
+import { matches } from "./util.js";
+
+
+/**
+ * Recursively execute a callback on this node and all its children.
+ * If the callback returns a non-undefined value, it will overwrite the node.
+ * This function will not modify the root node of the input AST.
+ * 
+ * @param {object | object[]} node AST node or array of nodes
+ * @param {function(object, string, object?)} transformCallback
+ * @param {object} [o]
+ * @param {string | string[] | function} [o.only] Only walk nodes of this type
+ * @param {string | string[] | function} [o.except] Ignore walking nodes of these types
+ */
+export default function transform (node, transformCallback, o) {
+	_transform(node, transformCallback, o);
+}
+
+function _transform (node, callback, o = {}, property, parent) {
+	if (Array.isArray(node)) {
+		return node.map(n => _transform(n, callback, o, property, parent));
+	}
+
+	const ignore = o.except && matches(node, o.except);
+	const explore = !ignore && matches(node, o.only);
+	
+	if (explore) {
+		const transformedNode = callback(node, property, parent);
+		if (transformedNode && node.parent) {
+			transformedNode.parent = node.parent;
+		}
```

Comment:
I would prefer we didn't introduce such a tight coupling with the parents module here. E.g. if we decide to switch from using simple properties to e.g. a WeakMap, we should not have to scour around unrelated modules to also tweak. Furthermore, modules should be able to ensure parent pointers are up to date without having to think much about logic.

We could detect if parent pointers exist before the transformation (btw `parent in node` is better for that, `node.parent` will return falsy even if it's null, which is still a valid value — but even better would be to offload that checking logic to the parents module as well) and then readjust them after the transformation by calling logic from the parents module (what could that look like? I have some ideas, but don't want to influence your creativity).

Reply:
I think a solution could be to add a `get` function to parents.js, which takes in a node and returns the parent if it exists and undefined otherwise.

Then this code could do something like
```js
...
const nodeParent = parents.get(node);
node = transformedNode !== undefined ? transformedNode : node;
parents.set(node, nodeParent);
...
```

Since this doesn't use the force option, it will only set the parent if the node doesn't already have it set. This solution eliminates all needs for `"parent" in node` or `node.parent` from `transform`. What do you think of this idea? Let me know, and I can create a separate PR for this if you think it's a good idea

Reply:
I really like it! `parents.get()` is nicely symmetrical with `parents.set()`. I do think we should use the `force` option however, since overriding an existing parent is very much the common case here (the node has a parent pointer to its previous parent and we want to change it to its new parent).